packages/rest/src/rest-http-error.ts

```diff
@@ -12,7 +13,25 @@ export namespace RestHttpErrors {
     const msg = `Parameters with "in: ${location}" are not supported yet.`;
     return new HttpErrors.NotImplemented(msg);
   }
-  export function invalidRequestBody(msg: string): HttpErrors.HttpError {
-    return new HttpErrors.UnprocessableEntity(msg);
+  export const INVALID_REQUEST_BODY_MESSAGE =
+    'The request body is invalid. See error object `details` property for more info.';
+  export function invalidRequestBody(): HttpErrors.HttpError {
+    return new HttpErrors.UnprocessableEntity(INVALID_REQUEST_BODY_MESSAGE);
   }
+  /**
+   * An invalid request body error contains a `details` property as the machine-readable error.
+   * Each entry in `error.details` contains the following attributes:
+   * - path: a path to the invalid field.
+   * - code: a single word code represents the error's type.
+   * - message: a humnan readable description of the error.
+   * - info: some additional details that the 3 attributes above don't cover.
+   * `ErrorDetails` defines the type of each entry, which is an object.
+   * The type of `error.details` is `ErrorDetails[]`
+   */
+  export type ErrorDetails = {
+    path: string;
+    code: string;
+    message: string;
+    info: object;
```

Comment:
Is the `info` (`params`) object already filled by AJV? Is it possible that AJV can return `undefined` for certain validation rules? In which case `info` would need to be marked as an optional property.

Reply:
ah...`params` is a required property, while `message` is not:
from ajv source code: 
```ts
  interface ErrorObject {
    keyword: string;
    dataPath: string;
    schemaPath: string;
    params: ErrorParameters;
    // Added to validation errors of propertyNames keyword schema
    propertyName?: string;
    // Excluded if messages set to false.
    message?: string;
    // These are added with the `verbose` option.
    schema?: any;
    parentSchema?: object;
    data?: any;
  }
```